src/Servers/Kestrel/Core/src/KestrelConfigurationLoader.cs

```diff
@@ -77,6 +77,7 @@ public class KestrelConfigurationLoader
     private IList<Action> EndpointsToAdd { get; } = new List<Action>();
 
     private CertificateConfig? DefaultCertificateConfig { get; set; }
+    internal X509Certificate2? DefaultCertificate { get; set; }
```

Comment:
Now stored here, rather than on `KestrelServerOptions` so that it's not mingled with the default store cert.

Reply:
This change caused a failure in CanReadAndWriteWithHttpsConnectionMiddlewareWithPemCertificate.
https://github.com/dotnet/aspnetcore/blob/33e7a9a111742c8b06924383baba5c448d7f8af8/src/Servers/Kestrel/test/InMemory.FunctionalTests/HttpsConnectionMiddlewareTests.cs#L54-L61
https://dev.azure.com/dnceng-public/public/_build/results?buildId=182674&view=ms.vss-test-web.build-test-results-tab&runId=3553510&resultId=122049&paneView=debug

Reply:
I'm assuming you meant to put this in the thread about untrusted certs?  Fixing that fixes this test.

Reply:
Is there a way to get notified (preferably, by email) when CI fails?  I keep thinking no new is good news.

Reply:
Not that I know of. @dougbu ?

Reply:
There is a way to enable notifications from AzDo for builds "you queued or were queued for you", I have this enabled so when I merge a PR I can see that the CI passes (or fails). It looks like queued builds from PRs don't count as "queued for you" though.

Reply:
I gave up on this a while ago. Suspect different AzDO teams got `Pull request by dougbu` working than left notifications broken. I'm testing things again in case soemthing somewhere was fixed recently&hellip;

Reply:
In a side-thread @wtgodbe and I hypothesized that we could have the bot comment when a CI failure occurs and then use the comment notification.

Reply:
Hmm, my change _didn't_ fix the test.  I'm investigating.

Reply:
`KestrelServerOptions` and `KestrelConfigurationLoader` are supposed to reference each other, but this test only makes the connection in one direction.

Reply:
Fixed.